torch/csrc/autograd/custom_function.cpp

```diff
@@ -26,6 +29,149 @@ Variable VariableInfo::zeros(at::OptionalDeviceGuard& device_guard) const {
   }
 }
 
+void _foreach_tensor(
+    const std::function<void (size_t, size_t, const at::Tensor&)>& fn,
```

Comment:
Since you are using this in a context where the body of this function is available inline, it is better to specify std::function as a lambda using templates. This can be done as

```
template <typename F>
void _foreach_tensor(F fn, ...)
```

this will inline the lambda and avoid creation on std::function